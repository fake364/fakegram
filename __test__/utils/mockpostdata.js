export default [
    {
        comments: [{author: "fake", "comment": "1231"}, {author: "fake", comment: "1231"}, {
            author: "fake",
            "comment": "hgj"
        }, {author: "fake", comment: "hgj"}, {author: "fake", comment: "hgj"}, {
            author: "fake",
            comment: "hgj"
        }, {author: "fake", comment: "hgj"}, {author: "fake", comment: "y"}, {
            "author": "fake",
            comment: "y"
        }, {author: "fake", comment: "y"}],
        description: "",
        image: "https://cdn.pixabay.com/photo/2020/04/23/03/35/the-leaves-5080909_960_720.jpg",
        author: {username: "test"}
    }
];