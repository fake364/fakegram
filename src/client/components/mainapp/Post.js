import React from 'react';
import {Link} from "react-router-dom";

function Post(props) {
    console.log(props);
    let username = props.post.author.username;
    return (
        <div className="Post">
            <header>
                <Link to={username}>
                    <img src="/dist/build/images/avatar.jpg" alt="Photo"/>
                </Link>
                <h2><Link to={username}>{username}</Link></h2>
                <img className="dots" src="/dist/build/images/dots.png"/>
            </header>
            <div className="postphoto">
                <img
                    src={props.post.image.includes("http") && props.post.image.includes("://") ? props.post.image : "images/posts/" + props.post.image}/>
            </div>
            <div className="FootBlock" style={{maxHeight: "300px"}}>
                <div className="LikeComments">
                    <div className="PostPanel">
                        <img className="Icon" alt="Photo" src="/dist/build/images/like.png"/>
                        <img className="Icon" alt="Photo" src="/dist/build/images/comment.png"/>
                    </div>
                    <h2>999 лайков</h2>
                    <div className="Comments" style={{
                        overflowY: "auto",
                        maxHeight: "150px"
                    }}>
                        {props.post.description !== "" ? <div>
                            <h2 style={{display: "inline-block", margin: "8px 0"}}>{username}</h2>
                            {" " + props.post.description}
                        </div> : null}
                        {props.post.comments.map((comment, index) =>
                            <div key={index}>
                                <h2 style={{display: "inline-block", margin: "8px 0"}}>{comment.author}</h2>
                                {" " + comment.comment}
                            </div>)
                        }

                    </div>
                </div>
                <div className="Comment">
                    <input required placeholder="Добавить комментарий...."/>
                    <button>Опубликовать</button>
                </div>
            </div>
        </div>

    );
}

export default Post;
