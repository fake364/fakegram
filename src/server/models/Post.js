const mongoose = require('mongoose');
const PostSchema = new mongoose.Schema({
    description: {type: String},
    author: {type: mongoose.Schema.Types.ObjectId, ref: "User"},
    image: {type: String},
    comments: {type: Array},
    date:{type:Date,default:Date.now}
});
module.exports = mongoose.model('Post', PostSchema);