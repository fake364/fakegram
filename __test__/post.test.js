import {expect, it} from "@jest/globals";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import Post from "../src/client/components/mainapp/Post";
import mockpostdata from "./utils/mockpostdata";
import React from "react";

it("should Post render correctly", () => {
    const element = renderer.create(<BrowserRouter><Post post={mockpostdata[0]}/></BrowserRouter>).toJSON();
    expect(element).toMatchSnapshot();
});