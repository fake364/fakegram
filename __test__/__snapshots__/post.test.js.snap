// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Post render correctly 1`] = `
<div
  className="Post"
>
  <header>
    <a
      href="/test"
      onClick={[Function]}
    >
      <img
        alt="Photo"
        src="/dist/build/images/avatar.jpg"
      />
    </a>
    <h2>
      <a
        href="/test"
        onClick={[Function]}
      >
        test
      </a>
    </h2>
    <img
      className="dots"
      src="/dist/build/images/dots.png"
    />
  </header>
  <div
    className="postphoto"
  >
    <img
      src="https://cdn.pixabay.com/photo/2020/04/23/03/35/the-leaves-5080909_960_720.jpg"
    />
  </div>
  <div
    className="FootBlock"
    style={
      Object {
        "maxHeight": "300px",
      }
    }
  >
    <div
      className="LikeComments"
    >
      <div
        className="PostPanel"
      >
        <img
          alt="Photo"
          className="Icon"
          src="/dist/build/images/like.png"
        />
        <img
          alt="Photo"
          className="Icon"
          src="/dist/build/images/comment.png"
        />
      </div>
      <h2>
        999 лайков
      </h2>
      <div
        className="Comments"
        style={
          Object {
            "maxHeight": "150px",
            "overflowY": "auto",
          }
        }
      >
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           1231
        </div>
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           1231
        </div>
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           hgj
        </div>
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           hgj
        </div>
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           hgj
        </div>
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           hgj
        </div>
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           hgj
        </div>
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           y
        </div>
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           y
        </div>
        <div>
          <h2
            style={
              Object {
                "display": "inline-block",
                "margin": "8px 0",
              }
            }
          >
            fake
          </h2>
           y
        </div>
      </div>
    </div>
    <div
      className="Comment"
    >
      <input
        placeholder="Добавить комментарий...."
        required={true}
      />
      <button>
        Опубликовать
      </button>
    </div>
  </div>
</div>
`;
